constructor/osxpkg.py

```diff
@@ -332,29 +331,25 @@ def move_script(src, dst, info, ensure_shebang=False, user_script_type=None):
     ).format(CHOSEN_PATH=f"$2/{pkg_name_lower}")
     # __osx is tested by the PKG metadata directly, no need to repeat
     virtual_specs = [spec for spec in info.get("virtual_specs", ()) if "__osx" not in spec]
-    replace = {
-        'NAME': info['name'],
-        'NAME_LOWER': pkg_name_lower,
-        'VERSION': info['version'],
-        'PLAT': info['_platform'],
-        'CHANNELS': ','.join(get_final_channels(info)),
-        'WRITE_CONDARC': '\n'.join(add_condarc(info)),
-        'PATH_EXISTS_ERROR_TEXT': path_exists_error_text,
-        'PROGRESS_NOTIFICATIONS': str(info.get('progress_notifications', False)),
-        'PRE_OR_POST': user_script_type or '__PRE_OR_POST__',
-        'CONSTRUCTOR_VERSION': info['CONSTRUCTOR_VERSION'],
-        'SHORTCUTS': shortcuts_flags(info),
-        'ENABLE_SHORTCUTS': str(info['_enable_shortcuts']).lower(),
-        'REGISTER_ENVS': str(info.get("register_envs", True)).lower(),
-        'VIRTUAL_SPECS': shlex.join(virtual_specs),
-        'NO_RCS_ARG': info.get('_ignore_condarcs_arg', ''),
-    }
-    data = preprocess(data, ppd)
-    custom_variables = info.get('script_env_variables', {})
-    data = fill_template(data, replace)
-
-    data = data.replace("_SCRIPT_ENV_VARIABLES_=''", '\n'.join(
-        [f"export {key}='{value}'" for key, value in custom_variables.items()]))
+    variables["pkg_name_lower"] = pkg_name_lower
+    variables["installer_name"] = info['name']
+    variables["installer_version"] = info['version']
+    variables["installer_platform"] = info['_platform']
+    variables["channels"] = ','.join(get_final_channels(info))
+    variables["write_condarc"] = '\n'.join(add_condarc(info))
```

Comment:
Simple join statements can be done by jinja, which I think that makes it more transparent. You may want to change that consistently throughout your template variables.

Reply:
I'd like to do a 2nd PR where we leverage Jinja features better. This PR would get too overwhelming otherwise, I feel.